### PR TITLE
Validate the key is in the dict

### PR DESCRIPTION
**Pull request by Forgind**
Fixes potential access problem.

### CODE REVIEW

File: src/Build/BackEnd/Components/Logging/LoggingService.cs [hunk 1]
```diff
@@ -546,9 +546,9 @@ public ICollection<string> GetWarningsToBeLoggedAsErrorsByProject(BuildEventCont
...
             }
 
-            if (_warningsAsMessagesByProject != null)
+            if (_warningsAsMessagesByProject != null && _warningsAsMessagesByProject.TryGetValue(key, out ISet<string> warningsAsMessages))
             {
-                warningsAsErrorsExcludingMessages.ExceptWith(_warningsAsMessagesByProject[key]);
+                warningsAsErrorsExcludingMessages.ExceptWith(warningsAsMessages);
 
                 // A non-null empty set means all warnings are errors. Avoid this.
                 if (warningsAsErrorsExcludingMessages.Count == 0)
```
Comment (rokonec) [inline src/Build/BackEnd/Components/Logging/LoggingService.cs:554]: This probably belongs to outer scope after if from line 549, as relying on _warningsAsErrorsByProject returning non empty set is a fragile promise.

**Review by rokonec — CHANGES_REQUESTED** [review]
Apart from that recommended change, which is not related to this commit, it is OK.

**Review by rokonec — APPROVED** [review]

**Review by KirillOsenkov — APPROVED** [review]
:+1:

**Comment by BenVillalobos** [comment]
This is strictly better than the previous version, approved.

Note that this method is easy to confuse with how warnings are actually converted, which is handled in `LogBuildEvent`, not here. Modifying this method only modifies how tasks understand that a warning they logged would be converted to an error.

Issues arise from a task relying on Log.HasLoggedError under this case:
1. All warnings are errors (warningsAsErrorsExcludingMessages will be a non-null empty set)
2. There are _any_ warningsAsMessages for this particular project.

Result: This method returns null, which implies that there are no warnings as errors. No task will see that a warning it logged was treated as an error. While this isn't great, it's the same functionality as what came before it. The workaround for this would be to manually select warnings to be errors. I'll file a bug on it.

**Review by BenVillalobos — APPROVED** [review]

**Pull request closed without merging**